seqr/utils/search/utils.py

```diff
@@ -103,21 +105,45 @@ def delete_search_backend_data(data_id):
         projects = set(active_samples.values_list('individual__family__project__name', flat=True))
         raise InvalidSearchException(f'"{data_id}" is still used by: {", ".join(projects)}')
 
-    return backend_specific_call(delete_es_index)(data_id)
+    return backend_specific_call(
+        delete_es_index, _raise_search_error('Deleting indices is disabled for the hail backend'),
+    )(data_id)
 
 
 def get_single_variant(families, variant_id, return_all_queried_families=False, user=None):
-    variants = backend_specific_call(get_es_variants_for_variant_ids)(
-        *_get_families_search_data(families), [variant_id], user, return_all_queried_families=return_all_queried_families,
+    variants = _get_variants_for_variant_ids(
+        families, [variant_id], user, return_all_queried_families=return_all_queried_families,
     )
     if not variants:
         raise InvalidSearchException('Variant {} not found'.format(variant_id))
     return variants[0]
 
 
 def get_variants_for_variant_ids(families, variant_ids, dataset_type=None, user=None):
-    return backend_specific_call(get_es_variants_for_variant_ids)(
-        *_get_families_search_data(families), variant_ids, user, dataset_type=dataset_type,
+    return _get_variants_for_variant_ids(families, variant_ids, user, dataset_type=dataset_type)
+
+
+def _get_variants_for_variant_ids(families, variant_ids, user, dataset_type=None, **kwargs):
+    parsed_variant_ids = {}
+    for variant_id in variant_ids:
+        try:
+            parsed_variant_ids[variant_id] = _parse_variant_id(variant_id)
+        except (KeyError, ValueError):
+            parsed_variant_ids[variant_id] = None
+
+    if dataset_type:
+        def is_valid(v_id):
+            if dataset_type == Sample.DATASET_TYPE_VARIANT_CALLS:
+                return bool(v_id)
+            return not v_id
```

Comment:
Will it be more neat to write these lines:
 `return bool(v_id) if dataset_type == Sample.DATASET_TYPE_VARIANT_CALLS else not v_id`